### PR TITLE
Preserve gradients during allocator tensor moves

### PR DESCRIPTION
**Pull request by FreddyE1982**
## Summary
- add a helper that preserves gradients when the resource allocator replaces tensor objects during transfers
- wire the gradient preservation helper through disk/GPU/CPU transfer paths, including fallbacks and restore helpers
- ensure the public `restore_tensor` utility keeps gradient buffers when moving tensors manually

## Testing
- python -m pytest tests/test_wanderer.py

------
https://chatgpt.com/codex/tasks/task_e_68c907307cf883279c4d289d16f4f8f5

### CODE REVIEW

File: marble/plugins/wanderer_resource_allocator.py [hunk 1]
```diff
@@ -772,6 +806,7 @@ def restore_tensor(obj: Any, attr: str, device: torch.device | str):
             moved = tensor.to(device)
             if getattr(tensor, "requires_grad", False) and not moved.requires_grad:
                 moved.requires_grad_(True)
+            moved = ResourceAllocatorPlugin._preserve_grad(tensor, moved)
             setattr(obj, attr, moved)
             return moved
         except Exception:
```
[the remark this answers is not in the document]
Comment (FreddyE1982) [reply marble/plugins/wanderer_resource_allocator.py:812]: @codex fix comments Gradients MUST be preserved!
